### PR TITLE
Add some useful methods to DataTable

### PR DESCRIPTION
**Pull request by klshrinidhi**
Adding some useful methods to `DataTable`. These methods can potentially be used in classes like `TableSource` which I am working on and is described in #889.

### CODE REVIEW

File: Bindings/common.i [hunk 1]
```diff
@@ -93,6 +94,13 @@
 %ignore OpenSim::AbstractDataTable::setIndependentMetaData;
 %ignore OpenSim::AbstractDataTable::getDependentsMetaData;
 %ignore OpenSim::AbstractDataTable::setDependentsMetaData;
+%ignore OpenSim::AbstractDataTable::setColumnLabels(
+                                     const std::initializer_list<std::string>&);
+%extend OpenSim::AbstractDataTable {
+    void setColumnLabels(const std::vector<std::string>& columnLabels) {
+        $self->setColumnLabels(columnLabels);
+    }
+}
```
Comment (chrisdembia) [inline Bindings/common.i]: +1

**Comment by chrisdembia** [comment]
Looks good to me. Just one minor comment: What do you think about rearranging the methods in AbstractDataTable so that the "column label" methods appear above the "metadata" methods in doxygen? Also, what about using doxygen groups to group all the "column label" methods, etc? Similar grouping could also be useful in DataTable (row accessors/mutators, column accessors/mutators, matrix/submatrix accessors/mutators).

File: OpenSim/Common/DataTable.h [hunk 2]
```diff
@@ -178,46 +178,175 @@ class AbstractDataTable {
...
     }
 
-    /** Set column labels.                                                    */
-    void setColumnLabels(const std::vector<std::string>& columnLabels) {
+    /** Get column label of a given column.                                   
+
+    \throws ColumnIndexOutOfRange If columnIndex is out of range of number of
+                                  columns.                                    
+    \throws KeyNotFound If column labels have not be set for the table.       */
+    const std::string& getColumnLabel(const size_t columnIndex) const {
```
Comment (aseth1) [inline OpenSim/Common/DataTable.h]: In this and the previous (and all column wise access) methods, It should be clearly documented that the columns indices, or iterate are only across the dependent columns of the table, with the independent column having its own accessors..

File: OpenSim/Common/DataTable.h [hunk 3]
```diff
@@ -341,38 +485,151 @@ class DataTable_ : public AbstractDataTable {
         return _indData;
     }
 
-    /** Get dependent column.                                                 
+    /** Get dependent column at index.
 
-    \throws ColumnIndexOutOfRange If index is out of range.                   */
+    \throws ColumnIndexOutOfRange If index is out of range for number of columns
+                                  in the table.                               */
     VectorView getDependentColumnAtIndex(size_t index) const {
```
Comment (aseth1) [inline OpenSim/Common/DataTable.h]: Maybe use "dependent" for column label accessors as well, to be consistent with accessing the data.

File: OpenSim/Common/DataTable.h [hunk 4]
```diff
@@ -341,38 +485,151 @@ class DataTable_ : public AbstractDataTable {
...
+    VectorView updDependentColumn(const std::string& columnLabel) {
+        return _depData.updCol(static_cast<int>(getColumnIndex(columnLabel)));
+    }
+
+    /** %Set independent column at index.                                      
 
     \throws RowIndexOutOfRange If index is out of range.                        
     \throws InvalidRow If this operation invalidates the row. Validation is
                        performed by derived classes.                          */
     void setIndependentColumnAtIndex(size_t index, const ETX& value) {
```
Comment (aseth1) [inline OpenSim/Common/DataTable.h]: I think this should be renamed to `setIndependentValueAtIndex` with a comment like: set the value at a specified index in the independent column of the `DataTable`.

Comment (chrisdembia) [inline OpenSim/Common/DataTable.h]: Maybe `index` should be named `rowIndex`. This is something that is generally confusing to me across the interface (when is the index a rowIndex or a colIndex?). I know that sometimes it only makes sense to be one or the other, but it takes some additional mental effort to realize that.

File: OpenSim/Common/DataTable.h (hunk 3 above)
Comment (chrisdembia) [inline OpenSim/Common/DataTable.h]: I'm fine with `getColumnLabels()`, etc., Using a name like `getDependentColumnLabels()` would then make it sound like there should be `getIndependentColumnLabel()`. Also, `DependentColumnLabel` is very long; if we are renaming methods, I'd want to get rid of `Dependent`. I think it's sufficient just to document `getColumnLabels()` saying that it does *not* include the independent column.

File: OpenSim/Common/DataTable.h (hunk 2 above)
Comment (chrisdembia) [inline OpenSim/Common/DataTable.h]: +1

**Comment by klshrinidhi** [comment]
* I have grouped the functions in Doxygen.
* Renamed the member function.
* Resolved merge conflict.

**Comment by klshrinidhi** [comment]
All :white_check_mark: !!!

**Comment by chrisdembia** [comment]
Looks good to me; but @aseth1 may have more to say?

**Comment by aseth1** [comment]
Yay! I would just add the comment about the column labels being just those of dependent columns in the comments for the column methods. Primarily because the current `Storage.getColumnLabels()` includes "time" which makes labels offset by 1 from the data (which has been so annoying!) and now they won't be!

**Comment by klshrinidhi** [comment]
@aseth1 
The group description for column labels mentions this explicitly. I didn't want to write the same comment for 3 or 4 functions related to column labels.

**Comment by klshrinidhi** [comment]
Sorry. I thought I added the description to mention this for column labels but I had done that for other groups.
I just push the change. 
Clear to merge ?

**Comment by aseth1** [comment]
Yes!

**Pull request merged**